src/createServer.js

```diff
@@ -2,7 +1,59 @@
+const http = require('http');
+const url = require('url');
+const fs = require('fs');
 
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    const normalizedUrl = new url.URL(req.url, `http://${req.headers.host}`);
+
+    const pathname = normalizedUrl.pathname.slice(1) || 'index.html';
+
+    res.setHeader('Content-Type', 'text/plain');
+
+    if (!req.url.startsWith('/file')) {
+      res.statusCode = 400;
+
+      res.end();
+
+      return;
+    }
+
+    if (!req.url.startsWith('/file/')) {
+      res.end('hint message');
+
+      return;
+    }
+
+    if (req.url.includes('//')) {
+      res.statusCode = 404;
+
+      res.end();
+
+      return;
+    }
+
+    const path = pathname.slice(5);
+
+    if (!fs.existsSync(`public/${path}`)) {
+      res.statusCode = 404;
+      res.end('File not found');
+
+      return;
+    }
+
+    fs.readFile(`public/${path}`, (err, data) => {
+      if (!err) {
+        res.end(data);
+
+        return;
+      }
```

Comment:
The error handling in the `fs.readFile` callback is not correctly implemented. Currently, it only checks if there is no error and proceeds to send the data. You should handle the error case first by checking if `err` is truthy, then set the appropriate status code and send an error message. This will ensure that any errors during file reading are properly handled.